### ISSUE TITLE
Support update/insert in save()

### ISSUE THREAD

**Issue by dhermes** [GoogleCloudPlatform/gcloud-python#599]
See https://github.com/GoogleCloudPlatform/gcloud-node/pull/387

**Comment by tseaver** [comment]
AFAICT, the "right" way to map those choices would be to add `datastore.insert()` and `datastore.update()`, methods, which would then have to expose the failure cases which `datastore.put()` elides via `upsert` (the only point in them would be to have the exceptions raised from `insert()` if entities exist already with the given keys, or from `update()` if no entities exist).

I'm not sure of the real utility of such methods from an application development perspective, but we could offer them for API-completeness sake.

**Comment by dhermes** [comment]
I don't even think they need to be top-level functions in `datastore.api`. We can just make them public methods on `Batch` so power users can use them. (And in the future we may want `insert` only behavior, specifically in a transaction. This was the original [report][1].)

[1]: https://github.com/GoogleCloudPlatform/gcloud-node/issues/384#issue-56782867

**Comment by dmcgrath** [comment]
Yes, this should definitely support insert, upsert, update as they are fundamental features of the database, not just power user features. For an example of how our other Cloud Datastore libraries to this, please review [Node.js](https://googlecloudplatform.github.io/google-cloud-node/#/docs/datastore/0.7.1/datastore?method=save) and [.NET](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/5c2c3e8e19f7ce0d533732164875e02d9897ba1d/apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreTransactionImpl.cs#L145)
